### ISSUE TITLE
Compilation error introduced in multi param Type classes

### ISSUE THREAD

**Issue by Abhiroop** [commercialhaskell/hindent#514]
When I run `hindent` on the following code:

```haskell
class (Num a, Num b) => ArithVecList a b where
  fold      :: (a -> a -> a) -> (b -> b -> b) -> b -> VecList b -> b
  zipVec    :: (a -> a -> a) -> (b -> b -> b) -> VecList b -> VecList b -> VecList b
```

This gets indented to:

```haskell
class (Num a, Num b) =>
      ArithVecList a b
  | 
  where
  fold :: (a -> a -> a) -> (b -> b -> b) -> b -> VecList b -> b
  zipVec ::
       (a -> a -> a) -> (b -> b -> b) -> VecList b -> VecList b -> VecList b
```

Notice the rogue `|` operator. This causes a compilation error.
```
hindent --version
hindent 5.2.4.1

ghc --version
The Glorious Glasgow Haskell Compilation System, version 8.2.2
```

**Comment by jgrosso** [comment]
I'm also experience this issue on hindent 5.2.4.1.

**Comment by jgrosso** [comment]
@Abhiroop Seems like this might've been fixed in 5.2.5, though? https://github.com/commercialhaskell/hindent/issues/482

**Comment by jgrosso** [comment]
Just upgraded to 5.2.7 and can confirm that it's resolved.
